stdlib/public/core/TemporaryAllocation.swift

```diff
@@ -84,10 +84,23 @@ internal func _isStackAllocationSafe(byteCount: Int, alignment: Int) -> Bool {
   // without worrying about running out of space, and the compiler would emit
   // such allocations on the stack anyway when they represent structures or
   // stack-promoted objects.
-  if byteCount <= 1024 {
+  if byteCount <= 1024 /* 1KiB */ {
     return true
   }
 
+  // Allocations larger than this limit are never candidates for stack promotion
+  // and will always be placed on the heap. If you're working on a platform with
+  // a particularly constrained stack limit, you might want to special-case this
+  // test and use a different upper bound.
```

Comment:
Can you add a TODO that says we should find a way to make this assert if a new platform is added and this isn't updated.

Also I haven't thought about how to do it, but it would be great to have a test.

Reply:
I don't think there's anything to test for!

Reply:
Added a TODO and an additional test to show that the emitted code just heap-allocates (haven't actually, you know, tested it though.)